src/coreclr/scripts/superpmi.py

```diff
@@ -1706,22 +1706,20 @@ async def create_one_artifact(jit_path: str, location: str, flags) -> str:
                             diff_txt = await create_one_artifact(self.diff_jit_path, diff_location, flags + diff_option_flags_for_diff_artifact)
 
                             if base_txt != diff_txt:
-                                jit_differences_queue.put_nowait(item)
+                                jit_differences_queue.put_nowait(context_index)
                     ################################################################################################ end of create_replay_artifacts()
 
-                    diff_items = []
-                    for item in self.diff_mcl_contents:
-                        diff_items.append(item)
-
                     logging.info("Creating dasm files: %s %s", base_asm_location, diff_asm_location)
-                    subproc_helper = AsyncSubprocessHelper(diff_items, verbose=True)
+                    dasm_contexts = self.pick_contexts_to_disassemble(diffs)
+                    subproc_helper = AsyncSubprocessHelper(dasm_contexts, verbose=True)
                     subproc_helper.run_to_completion(create_replay_artifacts, self, mch_file, asm_complus_vars_full_env, text_differences, base_asm_location, diff_asm_location, ".dasm")
 
                     if self.coreclr_args.diff_jit_dump:
                         logging.info("Creating JitDump files: %s %s", base_dump_location, diff_dump_location)
                         subproc_helper.run_to_completion(create_replay_artifacts, self, mch_file, jit_dump_complus_vars_full_env, jit_dump_differences, base_dump_location, diff_dump_location, ".txt")
 
-                    logging.info("Differences found. To replay SuperPMI use:")
+                    logging.info("Differences found. To replay SuperPMI use:".format(len(diffs)))
+
```

Comment:
What is this ".format..."? Shouldn't it use the logging '%s' format? And I don't see any replacement token.

Reply:
`logging` uses the same conventions as the old `%` formatting operator.  This is the newer [`str.format`](https://docs.python.org/3/library/stdtypes.html#str.format), which I think would normally be preferred... except that `logging` probably isn't going to format the string unless the particular logging level has been specified.

(and yes, it is missing `{}` or `{0}`)

Reply:
Whoops yes, at one point I was printing the number of diffs as part of this line, but I decided to print it on a new line at the end of each collection diff so that it gets printed last instead, and you can see it while waiting for the next diffs.

I'll keep in mind to remove this with my next change.

Reply:
It looks like logging can be switched to the new formatter: https://docs.python.org/3/library/logging.html#logging.Formatter

That would maybe be a nice cleanup at some point.

Reply:
Ah, looks like 'style' came with 3.2. Not sure what version of Python is on the CI systems, but presumably better than that.

Reply:
Yes, I (and some others) have been using new style formatting in various scripts for a while now without issues.